File: go/example_code/ses/ses_get_statistics.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package main

import (
	//go get -u github.com/aws/aws-sdk-go
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/ses"

	"fmt"
)

func main() {
	// Initialize a session that the SDK uses to load
	// credentials from the shared credentials file ~/.aws/credentials
	// and configuration from the shared configuration file ~/.aws/config.
	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	// Create an SES session.
	svc := ses.New(sess)

	// Attempt to send the email.
	result, err := svc.GetSendStatistics(nil)

	// Display any error message
	if err != nil {
		fmt.Println(err.Error())
		return
	}

	dps := result.SendDataPoints

	fmt.Println("Got", len(dps), "datapoints")
	fmt.Println("")

	for _, dp := range dps {
		fmt.Println("Timestamp: ", dp.Timestamp)
		fmt.Println("Attempts:  ", aws.Int64Value(dp.DeliveryAttempts))
		fmt.Println("Bounces:   ", aws.Int64Value(dp.Bounces))
		fmt.Println("Complaints:", aws.Int64Value(dp.Complaints))
		fmt.Println("Rejects:   ", aws.Int64Value(dp.Rejects))
		fmt.Println("")
	}
}

File: ruby/example_code/ses/ses_get_statistics.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX - License - Identifier: Apache - 2.0

# Purpose
# This code example demonstrates how to gets your Amazon Simple Email Service (Amazon SES) statistics.

# snippet-start:[s3.ruby.ses_get_statistics.rb]

require "aws-sdk-ses"  # v2: require 'aws-sdk'

# Create a new SES resource in the us-west-2 region.
# Replace us-west-2 with the AWS Region you're using for Amazon SES.
ses = Aws::SES::Client.new(region: "us-west-2")

begin
  # Get send statistics so we don't ruin our reputation
  resp = ses.get_send_statistics({})

  dps = resp.send_data_points

  puts "Got #{dps.count} data point(s):"
  puts

  dps.each do |dp|
    puts "Timestamp:  #{dp.timestamp}" #=> Time
    puts "Attempts:   #{dp.delivery_attempts}" #=> Integer
    puts "Bounces:    #{dp.bounces}" #=> Integer
    puts "Complaints: #{dp.complaints}" #=> Integer
    puts "Rejects:    #{dp.rejects}"  #-> Integer
    puts
  end

# If something goes wrong, display an error message.
rescue Aws::SES::Errors::ServiceError => error
  puts "Error: #{error}"
end
# snippet-end:[s3.ruby.ses_get_statistics.rb]
